src/Microsoft.HttpRepl.IntegrationTests/Mocks/MockedFileSystem.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.HttpRepl.FileSystem;
+using Microsoft.HttpRepl.IntegrationTests.Commands;
+
+namespace Microsoft.HttpRepl.IntegrationTests.Mocks
+{
+    internal class MockedFileSystem : IFileSystem
+    {
+        private readonly Dictionary<string, MockedFile> _files = new Dictionary<string, MockedFile>()
+        {
+            // Setup the "files" that should exist in the file system for the various tests here.
```

Comment:
Tests should set up the files they expect.  The mock should expose an API for priming the file system for tests.

Reply:
Maybe something fluent like:
```
var fileSystem = new MockFileSystem()
                  .AddFile(@"file/path", "content")
                  .AddFile(@"another/path", new byte[] { /* data */ })

Reply:
This is done. So far no tests need multiple files, but its set up as a Fluent call so it can be done as you described.